src/node.c

```diff
@@ -30,9 +30,6 @@ static bool S_can_contain(cmark_node *node, cmark_node *child) {
   if (node == NULL || child == NULL) {
     return false;
   }
-  if (NODE_MEM(node) != NODE_MEM(child)) {
```

Comment:
@vmg could you say more about why it would be bad to have nodes in the same tree that have different allocators?

> On Jun 8, 2016, at 12:49 PM, Vicent Martí <notifications@github.com> wrote:
> 
> In src/node.c:
> 
> > @@ -30,9 +30,6 @@ static bool S_can_contain(cmark_node *node, cmark_node *child) {
> >    if (node == NULL || child == NULL) {
> >      return false;
> >    }
> > -  if (NODE_MEM(node) != NODE_MEM(child)) {
> That's right. If you use cmark_node_new, you can only add that node to a document created from cmark_parser_new (since these APIs use the default system allocator)... Mixing and matching nodes is not a good idea. A library user should always use the same allocator, hence this check is important!
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>